library/timing.c

```diff
@@ -258,35 +81,27 @@ unsigned long mbedtls_timing_get_timer( struct mbedtls_timing_hr_time *val, int
     }
 }
 
-/* It's OK to use a global because alarm() is supposed to be global anyway */
-static DWORD alarmMs;
-
-static void TimerProc( void *TimerContext )
-{
-    (void) TimerContext;
-    Sleep( alarmMs );
-    mbedtls_timing_alarmed = 1;
-    /* _endthread will be called implicitly on return
-     * That ensures execution of thread funcition's epilogue */
-}
-
-void mbedtls_set_alarm( int seconds )
-{
-    if( seconds == 0 )
-    {
-        /* No need to create a thread for this simple case.
-         * Also, this shorcut is more reliable at least on MinGW32 */
-        mbedtls_timing_alarmed = 1;
-        return;
-    }
-
-    mbedtls_timing_alarmed = 0;
-    alarmMs = seconds * 1000;
-    (void) _beginthread( TimerProc, 0, NULL );
-}
-
 #else /* _WIN32 && !EFIX64 && !EFI32 */
 
+/**
```

Comment:
Possibly nicer to move the comment block and the first line of the function (i.e. the line "unsigned long mbedtls_timing_get_timer...") up outside of the #if defined ... blocks, so that it is clearly describing both implementations rather than just the non-WIndows case. This is very minor though.